synapse/lib/nexus.py

```diff
@@ -312,7 +314,10 @@ async def eat(self, nexsiden, event, args, kwargs, meta):
         if meta is None:
             meta = {}
 
-        return await self._eat((nexsiden, event, args, kwargs, meta))
+        async with self.applylock:
+            # Keep a reference to the shielded task to ensure it isn't GC'd
+            self.applytask = asyncio.create_task(self._eat((nexsiden, event, args, kwargs, meta)))
```

Comment:
Any perf measurements? Could we save on task construction by using a single task and a queue?

Reply:
sorry, just seeing your perf comment. reading... :D